velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -82,4 +82,13 @@ std::shared_ptr<exec::VectorFunction> makeRegexExtract(
   return result;
 }
 
+std::shared_ptr<exec::VectorFunction> makeRegexReplace(
+  const std::string& name,
+  const std::vector<exec::VectorFunctionArg>& inputArgs,
+  const core::QueryConfig& config) {
+    auto result = makeRe2Replace(name, inputArgs, config);
+    // ensureRegexIsCompatible("REGEXP_REPLACE", inputArgs[1].constantValue);
```

Comment:
Probably should enable this line.

Reply:
This line prevented passing vectors of regexes as the parameter. This was necessary in order to replicate the Unit test found here in Spark SQL: https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala#L180-L184

Reply:
The basic functionality of checking that the regex is compatible with RE2 is now handled within the functions themselves. Done once for every regex. Potential optimization would be to convert the passed regex vector into a map vector in order to prevent repeat calculations. Another optimization may be to just create a map and track if we have encountered a vector before and if it was compatible.